### PR TITLE
BUG: Fix infinite recursion observed from reading shaped_image

### PR DESCRIPTION
**Pull request by ronpandolfi**
As discussed in NIKEA: https://nikea.slack.com/archives/C7PNB3BUG/p1551207337008300

### CODE REVIEW

File: ophyd/areadetector/base.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def inverse(self, value):
         if not any(array_shape):
             raise RuntimeError(f"Invalid array size {self.derived_shape}")
 
-        array = self._derived_from.get()
+        array = self._derived_from.value
```
Comment (tacaswell) [inline ophyd/areadetector/base.py]: I think you want this to be `array = value` (the one that is passed in)

File: ophyd/areadetector/base.py [hunk 2]
```diff
@@ -105,8 +105,7 @@ def inverse(self, value):
         if not any(array_shape):
             raise RuntimeError(f"Invalid array size {self.derived_shape}")
 
-        array = self._derived_from.get()
-        return np.array(array).reshape(array_shape)
+        return np.array(value).reshape(array_shape)
```
Comment (ronpandolfi) [inline ophyd/areadetector/base.py]: A bit simpler, then

**Comment by teddyrendahl** [comment]
It wasn't you, but there is a rogue `print` statement in this function. Mind nixing it?

**Comment by tacaswell** [comment]
Calling `.read()` on it should be enough?

**Comment by mrakitin** [comment]
I guess so, reading a shaped image would be fine as it was the motivation for this PR.

**Pull request merged**